File: trunk/PO-8_210639/task_07/src/smilefacebuilder.cpp
#include "smilefacebuilder.h"
#include <qDebug>

void smileFaceBuilder::buildEars() {
	ears* newEars = new ears();
	newEars->color = "beidge";
	newEars->figure = "half circle";
	Ears = newEars;
}

void smileFaceBuilder::getResult() const
{
	qDebug() << "good face: ";
	Eyes->print();
	Ears->print();
	Mouth->print();
	Nose->print();
	Hair->print();
}

void smileFaceBuilder::buildEyes() {
	eyes* newEyes = new eyes();
	newEyes->color = "yellow";
	newEyes->figure = "circle";
	Eyes = newEyes;
}

void smileFaceBuilder::buildMouth() {
	mouth* newMouth = new mouth();
	newMouth->color = "red";
	newMouth->figure = "half circle";
	Mouth = newMouth;
}

void smileFaceBuilder::buildNose() {
	nose* newNose = new nose();
	newNose->color = "beidge";
	newNose->figure = "circle";
	Nose = newNose;
}

void smileFaceBuilder::buildHair() {
	hair* newHair = new hair();
	newHair->color = "brown";
	newHair->figure = "oval";
	Hair = newHair;
}

File: trunk/PO-8_210648/task_07/ConsoleApp1/ConsoleApp2/SmileFaceBuilder.cs
namespace ConsoleApp2;

public class SmileFaceBuilder : Builder
{
    private Product _product = new Product();

    public override void BuildEyes()
    {
        _product.AddEyes(new string[]{"smile","green"});
    }

    public override void BuildNose()
    {
        _product.AddNose(new string[]{"smile","green"});
    }

    public override void BuildMouth()
    {
        _product.AddMouth(new string[]{"smile","green"});
    }

    public override void BuildEars()
    {
        _product.AddEars(new string[]{"smile","green"});
    }

    public override void BuildHair()
    {
        _product.AddHair(new string[]{"smile","green"});
    }

    public override Product GetResult()
    {
        return _product;
    }
}